core/chains/evm/gas/arbitrum_estimator.go

```diff
@@ -0,0 +1,199 @@
+	})
+}
+func (a *arbitrumEstimator) Close() error {
+	return a.StopOnce("ArbitrumEstimator", func() (err error) {
+		close(a.chStop)
+		err = errors.Wrap(a.Estimator.Close(), "failed to stop gas price estimator")
+		<-a.chDone
+		return
+	})
+}
+
+// GetLegacyGas estimates both the gas price and the gas limit.
+func (a *arbitrumEstimator) GetLegacyGas(calldata []byte, l2GasLimit uint32, maxGasPriceWei *big.Int, opts ...Opt) (gasPrice *big.Int, chainSpecificGasLimit uint32, err error) {
+	gasPrice, _, err = a.Estimator.GetLegacyGas(calldata, l2GasLimit, maxGasPriceWei, opts...)
+	if err != nil {
+		return
+	}
+	ok := a.IfStarted(func() {
+		if slices.Contains(opts, OptForceRefetch) {
+			ch := make(chan struct{})
+			select {
+			case a.chForceRefetch <- ch:
+			case <-a.chStop:
+				err = errors.New("estimator stopped")
+				return
+			}
+			select {
+			case <-ch:
+			case <-a.chStop:
+				err = errors.New("estimator stopped")
+				return
+			}
+		}
+		perL2Tx, perL1CalldataUnit := a.getPricesInArbGas()
+		// TODO is it ok to return lsGasLimit if unitialized?
+		chainSpecificGasLimit = l2GasLimit + perL2Tx + uint32(len(calldata))*perL1CalldataUnit
+		a.logger.Debugw("GetLegacyGas", "l2GasLimit", l2GasLimit, "calldataLen", len(calldata), "perL2Tx", perL2Tx,
+			"perL1CalldataUnit", perL1CalldataUnit, "chainSpecificGasLimit", chainSpecificGasLimit)
+	})
+	if !ok {
+		return nil, 0, errors.New("estimator is not started")
+	} else if err != nil {
+		return
+	}
+	if max := a.cfg.EvmGasLimitMax(); chainSpecificGasLimit > max {
+		err = fmt.Errorf("estimated gas limit: %d is greater than the maximum gas limit configured: %d", chainSpecificGasLimit, max)
+		return
+	}
+	return
+}
+
+func (a *arbitrumEstimator) getPricesInArbGas() (perL2Tx uint32, perL1CalldataUnit uint32) {
+	a.getPricesInArbGasMu.RLock()
+	perL2Tx, perL1CalldataUnit = a.perL2Tx, a.perL1CalldataUnit
+	a.getPricesInArbGasMu.RUnlock()
+	return
+}
+
+func (a *arbitrumEstimator) run() {
+	defer close(a.chDone)
+
+	t := a.refreshPricesInArbGas()
+	close(a.chInitialised)
+
+	for {
+		select {
+		case <-a.chStop:
+			return
+		case ch := <-a.chForceRefetch:
+			t.Stop()
+			t = a.refreshPricesInArbGas()
+			close(ch)
+		case <-t.C:
+			t = a.refreshPricesInArbGas()
+		}
+	}
+}
+
+// refreshPricesInArbGas calls getPricesInArbGas() on the precompile contract 0x000000000000000000000000000000000000006c.
+func (a *arbitrumEstimator) refreshPricesInArbGas() (t *time.Timer) {
+	t = time.NewTimer(utils.WithJitter(a.pollPeriod))
+
+	ctx, cancel := evmclient.ContextWithDefaultTimeoutFromChan(a.chStop)
+	defer cancel()
+
+	// @return (per L2 tx, per L1 calldata unit, per storage allocation)
+	// function getPricesInArbGas() external view returns (uint256, uint256, uint256);
+	//
+	// https://github.com/OffchainLabs/nitro/blob/f7645453cfc77bf3e3644ea1ac031eff629df325/contracts/src/precompiles/ArbGasInfo.sol#L69
+	precompile := common.HexToAddress("0x000000000000000000000000000000000000006c")
```

Comment:
Possible to make the address as a config option?
We don't want to require new binary release if arbitrum ever wants to change this address.